csclient/csclient_test.go

```diff
@@ -185,6 +186,21 @@ func (s *suite) TestGet(c *gc.C) {
 }
 
 func (s *suite) TestGetArchive(c *gc.C) {
+	key := s.checkGetArchive(c)
+
+	// Check that the downloads count for the entity has been updated.
+	stats.CheckCounterSum(c, s.store, key, false, 1)
+}
+
+func (s *suite) TestGetArchiveWithStatsDisabled(c *gc.C) {
+	s.client.DisableStats()
+	key := s.checkGetArchive(c)
+
+	// Check that the downloads count for the entity has not been updated.
+	stats.CheckCounterSum(c, s.store, key, false, 0)
+}
```

Comment:
I think both of these tests would be more demonstrative if they started with a non-zero value.